nicegui/elements/plotly.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <div></div>
+</template>
+
+<script>
+export default {
+  mounted() {
+    setTimeout(() => {
+      this.ensureLibLoaded().then(() => {
+        // initial rendering of chart
+        Plotly.newPlot(this.$el.id, this.options.data, this.options.layout, this.options.config);
+
+        // register resize observer on parent div to auto-resize Plotly chart
+        let doResize = () => {
+          // only call resize if actually visible, otherwise error in Plotly.js internals
+          if (this.isHidden(this.$el)) return;
+          // console.log("Resize plot");
+          Plotly.Plots.resize(this.$el);
+        };
+
+        // throttle Plotly resize calls for better performance
+        // using HTML5 ResizeObserver on parent div
+        this.resizeObserver = new ResizeObserver((entries) => {
+          if (this.timeoutHandle) {
+            clearTimeout(this.timeoutHandle);
+          }
+          this.timeoutHandle = setTimeout(doResize, this.throttleResizeMs);
+        });
+        this.resizeObserver.observe(this.$el);
+      });
+    }, 0); // NOTE: wait for window.path_prefix to be set in app.mounted()
+  },
+  unmounted() {
+    this.resizeObserver.disconnect();
+    clearTimeout(this.timeoutHandle);
+  },
+
+  methods: {
+    isHidden(gd) {
+      // https://github.com/plotly/plotly.js/blob/e1d94b7afad94152db004b3bd5e6060010fbcc28/src/lib/index.js#L1278
+      var display = window.getComputedStyle(gd).display;
```

Comment:
I guess we can use `const` here.

Reply:
See https://github.com/zauberzeug/nicegui/pull/437/files#r1121745472